packages/components/src/select-control/styles/select-control-styles.ts

```diff
@@ -8,7 +8,7 @@ import styled from '@emotion/styled';
  * Internal dependencies
  */
 import { COLORS, rtl } from '../../utils';
-import type { Size } from '../types';
+import type { Size } from '../../input-control/types';
```

Comment:
What about keeping the reference to the internal component's types, so that we "expose" the dependency on `InputControl` only in `types.ts` ?

<details>

<summary>Something like this:</summary>

```diff
diff --git a/packages/components/src/select-control/styles/select-control-styles.ts b/packages/components/src/select-control/styles/select-control-styles.ts
index de35296fc0..7ffffaaaf9 100644
--- a/packages/components/src/select-control/styles/select-control-styles.ts
+++ b/packages/components/src/select-control/styles/select-control-styles.ts
@@ -8,11 +8,12 @@ import styled from '@emotion/styled';
  * Internal dependencies
  */
 import { COLORS, rtl } from '../../utils';
-import type { Size } from '../../input-control/types';
+import type { SelectControlProps } from '../types';
 
-interface SelectProps {
-	disabled?: boolean;
-	selectSize?: Size;
+// Using `selectSize` instead of `size` to avoid a type conflict with the
+// `size` HTML attribute of the `select` element.
+interface SelectProps extends Pick< SelectControlProps, 'disabled' > {
+	selectSize?: SelectControlProps[ 'size' ];
 }
 
 const disabledStyles = ( { disabled }: SelectProps ) => {
@@ -30,7 +31,7 @@ const fontSizeStyles = ( { selectSize }: SelectProps ) => {
 		'__unstable-large': '13px',
 	};
 
-	const fontSize = sizes[ selectSize as Size ];
+	const fontSize = sizes[ selectSize ?? 'default' ];
 	const fontSizeMobile = '16px';
 
 	if ( ! fontSize ) return '';
@@ -63,7 +64,7 @@ const sizeStyles = ( { selectSize }: SelectProps ) => {
 		},
 	};
 
-	const style = sizes[ selectSize as Size ] || sizes.default;
+	const style = sizes[ selectSize ?? 'default' ] || sizes.default;
 
 	return css( style );
 };
```
</details>

Reply:
Wonderful, I like that a lot better 👍  c026862